### DISCUSSION TITLE
Go to definition is not working in vue SFC components (*.vue files)

### DISCUSSION THREAD

**Discussion by FreshHead** [neovim/neovim#27891] [Q&A] [answered]
### Problem

Good day to you. 
I setup tsserver as described in configuration: 

https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#vue-support

But when I use "go to definition" with cursor under some custom component I always end up in shims-vue.d.ts file of my project.

Probably that is out of neovim scope. But maybe there is a smallest chance that configuration example can be updated somehow.

My neovim config is based on kickstart.nvim. Below is the part made by instructions in https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#vue-support:

```lua
        tsserver = {
          filetypes = {
            'typescript',
            'vue',
          },
          init_options = {
            plugins = {
              {
                name = '@vue/typescript-plugin',
                location = '/home/u/.nvm/versions/node/v20.8.0/lib/node_modules/@vue/typescript-plugin',
                languages = { 'javascript', 'typescript', 'vue' },
              },
            },
          },
        },

        volar = { filetypes = { 'vue' } },
```
Than I use `lua vim.lsp.buf.definition()`

### Expected behavior

The file with my component is opened

### Neovim version (nvim -v)

0.9.5

### Language server name/version

typescript-language-sever 4.3.3,  vue-language-server 2.0.6

### Operating system/version

linux mint 21

### Log file

https://gist.github.com/FreshHead/64b96748af826e3f8c0406561c4b4cd6

**Comment by SeanSelene** [comment, marked as answer]
which version? Volar's go to definition doesn't work for vue2.
In addition，the location should be the path of volar otherwise it doesn't work for me.
```lua
local data_path = vim.fn.stdpath('data')
local location = data_path.."/mason/packages/vue-language-server/node_modules/@vue/language-server"

return {
  init_options = {
    plugins = {
      {
        name = "@vue/typescript-plugin",
        location = location,
        languages = {"javascript", "typescript", "vue"},
      },
    },
  },
  filetypes = {
    "javascript",
    "typescript",
    "vue",
  },
}
```

**Reply by FreshHead** [reply]
I was definitely wrong with location. Thank you. I tried to change location but it did not help (
 
 My versions:
 "vue": "3.2.47",
 "@vue/cli-service": "~5.0.8",
 
I think something wrong in project configuration. When I am checking out Vite migration branch go to definition is working as expected. Therefore I close this issue
